src/cmd/cmd.zig

```diff
@@ -405,35 +412,30 @@ fn validator() !void {
         entrypoints,
         ip_echo_data.shred_version, // TODO atomic owned at top level? or owned by gossip is good?
         ip_echo_data.ip,
-        &.{.{ .tag = socket_tag.REPAIR, .port = repair_port }},
+        &.{
+            .{ .tag = socket_tag.REPAIR, .port = repair_port },
+            .{ .tag = socket_tag.TVU, .port = tvu_port },
+        },
     );
     defer gossip_service.deinit();
     var gossip_handle = try spawnGossip(&gossip_service);
 
-    // repair
-    var repair_socket = try Socket.create(network.AddressFamily.ipv4, network.Protocol.udp);
-    try repair_socket.bindToPort(repair_port);
-    try repair_socket.setReadTimeout(SOCKET_TIMEOUT);
-
-    var repair_svc = try initRepair(
-        logger,
-        &my_keypair,
-        &exit,
-        rand.random(),
-        &gossip_service,
-        &repair_socket,
-    );
-    defer repair_svc.deinit();
-    var repair_handle = try std.Thread.spawn(.{}, RepairService.run, .{&repair_svc});
-
-    var shred_receiver = ShredReceiver{
+    // shred collector
+    var shred_collector = try sig.shred_collector.start(.{
```

Comment:
What do you mean by "all in one go"? Do you mean using a single struct instead of three? Or are you also advocating for using function parameters instead of structs?

It may be best to merge the dependencies struct with the interface struct, because the distinction between those ideas is pretty fuzzy. It's probably easier to just call them all dependencies. So that would narrow it down to two structs.

I like to keep config and dependencies separate because they are conceptually distinct and need to be handled differently. Separating them into different structs makes the distinction more obvious. To me this makes the code easier to read because the code is self-documenting in a way. But I am biased since I'm the one who wrote it.

Also, it's common to instantiate config and dependencies in different places. It's easier to do that if they are separate structs.

In terms of function params vs a struct, when you have a large number of function parameters, it gets hard to make sense of them at call sites. I find it easier to read and work with named fields in the struct syntax. Also, structs package things up neatly and make it easy to collect everything and move it around before calling the function.

> one since its easier to read/one less click-through link to see what the params are

The structs are defined right next to each other and the function, so you can still look at it all at once.

Reply:
597677684d65918748040459967ef1626b4cb18a

Reply:
makes sense

i like the named fields in the struct too - im not sure where we would draw the line for this approach though - ie, what functions should take structs and what should take parameters - is the readability improvement worth it vs the consistency throughtout in the codebase (because some fcns will use structs others wont)

i think using this approach here is ok for now - but i also think its worth the thought of if/how we want to approach this going forward - would be interested in what @InKryption thinks too

Reply:
chatted offline on this - ok with using (inline) structs for function parameters - in general:
	- too many parameters in one fcn a struct helps organizaiton
	- multiple parameters of same type - struct with explicit naming helps
	- ok with using inline structs for this purpose
		- if we only use it once maybe not worth vs using it more often